### PR TITLE
📚 DOCS: Execute graph tutorial as Jupyter Notebook

### PR DESCRIPTION
**Pull request by chrisjsewell**
This PR adds the [myst-nb](https://github.com/executablebooks/MyST-NB) dependency, in order to execute the "How to visualize provenance" as a Jupyter Notebook.
It uses the new features from #5375, to load the archive as a profile, then code proceeds with the standard aiida API, i.e. no need to for postgresql 😄 

See: https://aiida--5422.org.readthedocs.build/projects/aiida-core/en/5422/howto/visualising_graphs/visualising_graphs.html

This has two important benefits:
1. Since the code is actually being executed (and will fail the build if it excepts), therefore it is validating that it actually works
2. Users can download the notebook to run directly (and you could even add Binder/thebe later, to run in-browser) 

Note:
- The archive is stored in the docs unzipped. This means that the sqlite database does not have to be extracted to a temporary file, when loading the archive as a profile
- The notebook is stored in the [MyST Markdown format](https://jupytext.readthedocs.io/en/latest/formats.html#myst-markdown), this is better for editing, reviewing and for git diffs.

### CODE REVIEW

**Comment by chrisjsewell** [comment]
this is a bit of a proof-of-principle, for using it also elsewhere in the documentation 😉

**Comment by chrisjsewell** [comment]
FYI, the `SqliteZipBackend` is currently "read-only", i.e. you cannot add data to it, and thus cannot run calculations with the profile.
But, in its unzipped form, there is actually no reason why you could not write to it (i.e. modify the sqlite database and/or add repo files). Obviously, you wouldn't want to use this for production runs (mainly due to concurrency of the sqlite database), but I think that would be useful for tutorials on AiiDA process execution (both here and in plugins)

**Comment by chrisjsewell** [comment]
@sphuber, now re-based

File: docs/source/howto/visualising_graphs/graph1.aiida/repo/8f434346648f6b96df89dda901c5176b10a6d83961dd3c1ac88b59b2dc327aa4 [hunk 1]
```diff
@@ -0,0 +1 @@
+hi
```
Comment (sphuber) [inline docs/source/howto/visualising_graphs/graph1.aiida/repo/8f434346648f6b96df89dda901c5176b10a6d83961dd3c1ac88b59b2dc327aa4:1]: what's this file doing here? Is this just a placeholder? I take it it is just one of the nodes in the `graph1.aiida` has this file, but it seems kind of pointless for the graph tutorial. Might as well get rid off it.

**Review by sphuber — COMMENTED** [review]
You say that an error in the nb will fail the build, but what currently triggers the execution?

File: docs/source/howto/visualising_graphs/graph1.aiida/repo/8f434346648f6b96df89dda901c5176b10a6d83961dd3c1ac88b59b2dc327aa4 (hunk 1 above)
Comment (chrisjsewell) [reply docs/source/howto/visualising_graphs/graph1.aiida/repo/8f434346648f6b96df89dda901c5176b10a6d83961dd3c1ac88b59b2dc327aa4:1]: Yeh no worries, just an artifact of when I created this archive many years ago

**Comment by mbercx** [comment]
Looks great, thanks @chrisjsewell ! Just one question: Do we have to provide the `graph1.aiida` archive or should that be created when creating the profile?

**Comment by chrisjsewell** [comment]
> You say that an error in the nb will fail the build, but what currently triggers the execution?

It's always executed on a new docs build (or when the file changes on a docs rebuild), for more information see: https://myst-nb.readthedocs.io/en/latest/use/execute.html

**Comment by chrisjsewell** [comment]
> Do we have to provide the graph1.aiida archive or should that be created when creating the profile?

This code is being run against a "pre-built" archive, at `docs/source/howto/visualising_graphs/graph1.aiida`.
If you wanted to run the notebook locally, then you would need to also have that archive local.

File: docs/source/howto/visualising_graphs/graph1.aiida/repo/8f434346648f6b96df89dda901c5176b10a6d83961dd3c1ac88b59b2dc327aa4 (hunk 1 above)
Comment (sphuber) [reply docs/source/howto/visualising_graphs/graph1.aiida/repo/8f434346648f6b96df89dda901c5176b10a6d83961dd3c1ac88b59b2dc327aa4:1]: Alright, if you remove that file from the archive I will approve

**Review by sphuber — APPROVED** [review]

**Pull request merged**